src/wallet/asyncrpcoperation_mergetoaddress.cpp

```diff
@@ -0,0 +1,926 @@
+
+    size_t numInputs = utxoInputs_.size();
+
+    // Check mempooltxinputlimit to avoid creating a transaction which the local mempool rejects
+    size_t limit = (size_t)GetArg("-mempooltxinputlimit", 0);
+    if (limit > 0 && numInputs > limit) {
+        throw JSONRPCError(RPC_WALLET_ERROR,
+                           strprintf("Number of transparent inputs %d is greater than mempooltxinputlimit of %d",
+                                     numInputs, limit));
+    }
+
+    CAmount t_inputs_total = 0;
+    for (MergeToAddressInputUTXO& t : utxoInputs_) {
+        t_inputs_total += std::get<1>(t);
+    }
+
+    CAmount z_inputs_total = 0;
+    for (MergeToAddressInputNote& t : noteInputs_) {
+        z_inputs_total += std::get<2>(t);
+    }
+
+    CAmount targetAmount = z_inputs_total + t_inputs_total;
+
+    if (targetAmount <= minersFee) {
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS,
+                           strprintf("Insufficient funds, have %s and miners fee is %s",
+                                     FormatMoney(targetAmount), FormatMoney(minersFee)));
+    }
+
+    CAmount sendAmount = targetAmount - minersFee;
+
+    // update the transaction with the UTXO inputs and output (if any)
+    CMutableTransaction rawTx(tx_);
+    for (MergeToAddressInputUTXO& t : utxoInputs_) {
+        CTxIn in(std::get<0>(t));
+        rawTx.vin.push_back(in);
+    }
+    if (isToTaddr_) {
+        CScript scriptPubKey = GetScriptForDestination(toTaddr_.Get());
+        CTxOut out(sendAmount, scriptPubKey);
+        rawTx.vout.push_back(out);
+    }
+    tx_ = CTransaction(rawTx);
+
+    LogPrint(isPureTaddrOnlyTx ? "zrpc" : "zrpcunsafe", "%s: spending %s to send %s with fee %s\n",
+             getId(), FormatMoney(targetAmount), FormatMoney(sendAmount), FormatMoney(minersFee));
+    LogPrint("zrpc", "%s: transparent input: %s\n", getId(), FormatMoney(t_inputs_total));
+    LogPrint("zrpcunsafe", "%s: private input: %s\n", getId(), FormatMoney(z_inputs_total));
+    if (isToTaddr_) {
+        LogPrint("zrpc", "%s: transparent output: %s\n", getId(), FormatMoney(sendAmount));
+    } else {
+        LogPrint("zrpcunsafe", "%s: private output: %s\n", getId(), FormatMoney(sendAmount));
+    }
+    LogPrint("zrpc", "%s: fee: %s\n", getId(), FormatMoney(minersFee));
+
+    // Grab the current consensus branch ID
+    {
+        LOCK(cs_main);
+        consensusBranchId_ = CurrentEpochBranchId(chainActive.Height() + 1, Params().GetConsensus());
+    }
+
+    /**
+     * SCENARIO #1
+     *
+     * taddrs -> taddr
+     *
+     * There are no zaddrs or joinsplits involved.
+     */
+    if (isPureTaddrOnlyTx) {
+        UniValue obj(UniValue::VOBJ);
+        obj.push_back(Pair("rawtxn", EncodeHexTx(tx_)));
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #1
+     */
+
+
+    // Prepare raw transaction to handle JoinSplits
+    CMutableTransaction mtx(tx_);
+    if (!mtx.fOverwintered) {
```

Comment:
Remove.  Rationale is that the caller in rpcwallet instantiates this async operation passing in a mutable tx which is the template to build on.  Format and contextual data 
 e.g. nVersion (and versionGroupId etc) are set by the caller.